src/compile.ts

```diff
@@ -236,14 +248,22 @@ export function compile(
             // if this is a reference to a Table or Lambda, retain it
             return ref(node);
           }
+          const type = checker.getTypeAtLocation(node.name);
           return newExpr("PropAccessExpr", [
             toExpr(node.expression),
             ts.factory.createStringLiteral(node.name.text),
+            type
+              ? ts.factory.createStringLiteral(checker.typeToString(type))
+              : ts.factory.createIdentifier("undefined"),
           ]);
         } else if (ts.isElementAccessExpression(node)) {
+          const type = checker.getTypeAtLocation(node.argumentExpression);
           return newExpr("ElementAccessExpr", [
             toExpr(node.expression),
             toExpr(node.argumentExpression),
+            type
+              ? ts.factory.createStringLiteral(checker.typeToString(type))
```

Comment:
Should we convert this to an object instead of a string?

Reply:
Yeah, probably. I was just happy to get some type data at all :-). I also don't know if its the right value, this returns `string[]`, `number[]`, `boolean`. An object would allow us to add more forms of the type later, so object seems right.

Reply:
What do we need the types for? We could convert types to Json schema. Typescript's structural typing maps well.

Reply:
Just need to know the type of the property being interacted with. 

`bool` https://github.com/sam-goodwin/functionless/pull/39/files#diff-39df29cf96e7006909d1516b2738e52ab55c0e7e65f10a5e2dee5be1dc96157eR304-R309
`*[]` to know if includes should work
`number` to do numeric ranges.

Reply:
Cool, we can start with this.